packages/extension/background.js

```diff
@@ -50,14 +50,50 @@ chrome.contextMenus.onClicked.addListener(async (info, tab) => {
 });
 
 chrome.action.onClicked.addListener((tab) => {
-  chrome.scripting.executeScript({
-    target: { tabId: tab.id, allFrames: false },
-    func: createPanel,
-  });
+  getEmailCheck(tab)
+  .then((email) => {
+    chrome.scripting.executeScript({
+      target: { tabId: tab.id, allFrames: false },
+      args: [email],
+      func: createPanel,
+    });
+  })
 });
 
 chrome.runtime.onMessage.addListener((message) => {
   if (message.type === "save-prefs") {
   } else if (message.type === "get-defs") {
   }
 });
+
+
+const getEmailCheck = (tab) => {
+  return new Promise((resolve, reject) => {
+    // Get email if we know where to look
+    if (tab.url.includes("zendesk")) {
+      chrome.scripting.executeScript({
+        target: { tabId: tab.id, allFrames: false },
+        func: () => {
+          let email = "";
+          const userEmailElement = document.getElementsByClassName("email-text")
+          if (userEmailElement.length > 0) {
+            email = userEmailElement[0].innerText
+          }
+
+          if (!email) {
+            const ticketEmail = document.getElementsByClassName("sc-7acs88-3 hljQCC")
+            if (ticketEmail.length > 0) {
+              email = ticketEmail[0].innerText
+            }
+          }
+          return email
+        },
+      }).then((result) => {
+        let email = result[0].result || "";
+        return resolve(email);
+      });
+    } else {
+      return resolve("");
```

Comment:
This will probably pull up a random list of people since we're setting the search param to `?email=`. It would be better to call `reject` here, and then catch it in the onClicked handler.